### PR TITLE
Implemented eraser

### PR DESCRIPTION
**Pull request by Eirmas**
Rewrote all tools to use the Konva.Group class. Did this because one tool can consist of multiple shapes, so its nice to keep them grouped. The groups also made it a lot easier to implement the eraser

### CODE REVIEW

**Comment by Eirmas** [comment]
Eraser works both by clicking and by draging. Small objects can be a little hard to hit, but still works
Eraser Example

File: src/tools/Erase.ts [hunk 1]
```diff
@@ -1,37 +1,58 @@
...
 
-  // eslint-disable-next-line
+  findAndEraseGroup = (e: Konva.KonvaPointerEvent, canvasElement: CanvasElement, layer: Konva.Layer, socket: WebSocket): void => {
+    const groupId = e.target.parent?.attrs.id
+    this.eraseGroup(canvasElement, layer, groupId)
+    this.sendToWebSocket(canvasElement, socket)
+  }
+
+  eraseGroup = (canvasElement: CanvasElement, layer: Konva.Layer, groupId?: string): void => {
+    if (groupId !== undefined && groupId !== null) {
```
Comment (bryansamuel1) [inline src/tools/Erase.ts]: !groupId would be better?

File: src/tools/Erase.ts [hunk 2]
```diff
@@ -1,37 +1,58 @@
...
+      const group = layer.getChildren(node => {
+        return node.attrs.id === groupId
```
Comment (bryansamuel1) [inline src/tools/Erase.ts]: Do not need separate lines for return. Remove curly braces and return keyword then put on one Line

**Comment by bryansamuel1** [comment]
Eirmas, see comments in discord for further work on this component. It is fine to merge this ticket into develop but you need to create a new ticket

**Pull request closed without merging**